### PR TITLE
[WIP] Add node log query documentation

### PR DESCRIPTION
**Pull request by LorbusChris**
Draft for adding documentation for the Node service log viewer feature

- Kubernetes Enhancement Proposal: [KEP-2258: Node service log viewer](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/2258-node-service-log-viewer)
- KEP Discussion: https://github.com/kubernetes/enhancements/issues/2258
- Code: https://github.com/kubernetes/kubernetes/pull/96120

### CODE REVIEW

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 1]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
 the `kube-up.sh` script, those logs are configured to be rotated by
 the `logrotate` tool daily or once the size exceeds 100MB.
 
+## Node Log Viewer
+
+{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
+
+Kubernetes v1.24 offers initial support for viewing node logs using kubectl;
+this is an alpha feature that is disabled by default. You can enable the alpha
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/cluster-administration/logging.md:140]: Isn't the fact that an alpha feature is turned off by default commonplace knowledge now?

```suggestion
This is an alpha feature and can be enabled
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 2]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
 the `kube-up.sh` script, those logs are configured to be rotated by
 the `logrotate` tool daily or once the size exceeds 100MB.
 
+## Node Log Viewer
+
+{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
+
+Kubernetes v1.24 offers initial support for viewing node logs using kubectl;
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/cluster-administration/logging.md:139]:
```suggestion
Kubernetes v1.24 offers initial support for viewing node logs using kubectl.
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 3]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
...
+this is an alpha feature that is disabled by default. You can enable the alpha
+feature by turning on the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/cluster-administration/logging.md:141]:
```suggestion
by turning on the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
```

**Review by divya-mohan0209 — COMMENTED** [review]
Slight nit + rephrasing to avoid redundancy since we're linking to the feature gates documentation

**Comment by sftim** [comment]
Is this still part of the v1.24 release? If not, I recommend unsetting the milestone.

**Comment by aravindhp** [comment]
This is not part of the 1.24 release.

**Comment by mehabhalodiya** [comment]
Hi @LorbusChris, I don't think this PR should target `dev-1.24`.

**Comment by Sea-n** [comment]
Hey @LorbusChris @aravindhp,

There are some comments you haven't replied yet.

Are you ready to move forward?

**Comment by aravindhp** [comment]
@Sea-n I will address those comments once we are done with https://github.com/kubernetes/kubernetes/pull/96120

**Comment by sftim** [comment]
The PR description says that these changes are in draft.
/retitle [WIP] Add node log viewer documentation

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 4]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
 the `kube-up.sh` script, those logs are configured to be rotated by
 the `logrotate` tool daily or once the size exceeds 100MB.
 
+## Node Log Viewer
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:135]:
```suggestion
## Node log viewer
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 5]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
...
+
+{{< feature-state for_k8s_version="v1.25" state="alpha" >}}
+
+Kubernetes v1.25 offers initial support for viewing node logs using kubectl.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:139]:
```suggestion
Kubernetes v{{< skew currentVersion >}} includes initial support for viewing node logs using `kubectl`.
```

That way it's right for each subsequent release. The feature-state shortcode tells readers what version provides this support (and if that's not clear, we should improve the shortcode).

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 6]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
...
+This is an alpha feature and can be enabled
+by turning on the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+named `NodeLogViewer` for the kubelet component.
+
+With the feature enabled you can use `kubectl alpha node-logs` to view the logs
+of system services and log files in `/var/log/` on both Linux and Windows nodes.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:145]: - Does Windows have a `/var/log`
- We don't know what path the kubelet is writing logs to.

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 7]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
...
+
+Example usage:
+```
+  # Show kubelet logs from all worker nodes
+  kubectl alpha node-logs --role worker -u kubelet
+
+  # See what logs are available in a worker node called w1 in /var/logs
+  kubectl alpha node-logs w1 --path=/
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:153]: - Don't indent these commands
- We don't know what path the kubelet is writing logs to (it's _usually_ `/var/log`)

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 8]
```diff
@@ -132,6 +132,29 @@ directory should be rotated. In Kubernetes clusters brought up by
...
+```
+
+More details can be found in the kubectl help i.e. `kubectl alpha node-logs --help`.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md:156]:
```suggestion
You can learn more about this subcommand by running
`kubectl alpha node-logs --help`.
```

**Review by sftim — COMMENTED** [review]
Thanks for this P @LorbusChris 

I'd put most of this text into a short new page within https://kubernetes.io/docs/tasks/debug/debug-cluster/

For https://kubernetes.io/docs/concepts/cluster-administration/logging/ I would put a heading and a paragraph of text that links to the task page.

Does that make sense?

**Comment by reylejano** [comment]
/assign @cathchu 
/cc @kcmartin 
continue hold as the related enhancement did not make 1.25 and SIG Windows is targeting 1.26 for this enhancement
see https://github.com/kubernetes/enhancements/issues/2258#issuecomment-1203014627

**Comment by reylejano** [comment]
@LorbusChris please target the `main` branch and when the `dev-1.26` branch is created (shortly after the 1.25 release) you can change the target to `dev-1.26`
[Here's an example from GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request)
/milestone clear

**Comment by krol3** [comment]
> @LorbusChris please target the `main` branch and when the `dev-1.26` branch is created (shortly after the 1.25 release) you can change the target to `dev-1.26` [Here's an example from GitHub](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-base-branch-of-a-pull-request) /milestone clear

@reylejano  This KEP was removed of the release 1.26. cc: @cathchu

**Pull request closed without merging**